### ISSUE TITLE
`docker login` (using `--insecure-registry`) fails with `401 Unauthorized` if you change your password

### ISSUE THREAD

**Issue by vikstrous** [docker/docker#17882]
Steps to reproduce:

1. Create an account with a registry (it should work with docker hub the same way)
2. Log in with `docker login`
3. Change your password
4. If you try to push it fails, so try to log in again by running `docker login`

**Expected**: you can log in with your new password
**Actual**: Cryptic 401 error and no way to actually change the stored password
**Workaround**:
```
$ docker logout
$ docker login
```

Examples:
```
$ docker login 172.17.42.1
Username (admin): 
Error response from daemon: no successful auth challenge for https://172.17.42.1/v2/ - errors: [token auth attempt for registry https://172.17.42.1/v2/: https://172.17.42.1/auth/token?account=admin&service=172.17.42.1 request failed with status: 401 Unauthorized]
```
```
$ docker login 172.17.42.1
Username (admin): admin
Error response from daemon: no successful auth challenge for https://172.17.42.1/v2/ - errors: [token auth attempt for registry https://172.17.42.1/v2/: https://172.17.42.1/auth/token?account=admin&service=172.17.42.1 request failed with status: 401 Unauthorized]
```

Docker version used:
```
$ docker version
Client:
 Version:      1.9.0
 API version:  1.21
 Go version:   go1.5.1
 Git commit:   76d6bc9-dirty
 Built:        Thu Nov  5 20:07:10 UTC 2015
 OS/Arch:      linux/amd64

Server:
 Version:      1.9.0
 API version:  1.21
 Go version:   go1.5.1
 Git commit:   76d6bc9-dirty
 Built:        Thu Nov  5 20:07:10 UTC 2015
 OS/Arch:      linux/amd64
```

**Comment by askb** [comment]
If you are using local registry and the endpoint has been setup with a list of authorization challenges this error should not be seen. If you are connecting docker daemon to local registry using `--insecure-registry` this could be the cause of the issue. Looks like this needs to be properly handled in `registry.Loginv2()`.

**Comment by vikstrous** [comment]
Ah, yes, I was using `--insecure-registry`. I assumed it would work the same way for Hub. It looks like it doesn't. When I tried it on Hub it logged me out after one failed login attempt.

It would be nice if running `docker login` never tried to use your old credentials. I don't understand why it does that at all.

**Comment by askb** [comment]
I am facing a similar issue. once the on-disk credentials are invalidated, user authentication must be re-attempted (at-least three times) but the second time, user must be prompted to input the credentials along with a meaningful message, "invalid credential cache".     

Also noticed another issue when running `docker daemon` with `--disable-legacy-registry=true` `--insecure-registry`, could be related to the same issue.

```
$ docker login
Username: admin
Password: 
Email: admin@....com
Error response from daemon: invalid registry endpoint https://index.docker.io/v2/: https://index.docker.io/v2/ does not appear to be a v2 registry endpoint. If this private registry supports only HTTP or HTTPS with an unknown CA certificate, please add `--insecure-registry index.docker.io` to the daemon's arguments. In the case of HTTPS, if you have access to the registry's CA certificate, no need for the flag; simply place the CA certificate at /etc/docker/certs.d/index.docker.io/ca.crt 
```

**Comment by vikstrous** [comment]
You should probably file a separate github issue for that. I can reproduce it. It seems like the `--disable-legacy-registry=true` fails regardless of whether or not you have `--insecure-registry` set.

**Comment by aaronlehmann** [comment]
This appears to have been fixed in 1.11. I was able to change my password on Docker Hub and run `docker login` with no problems. It now prompts for a password every time you run `docker login` instead of trying to reuse the old one.
